cmd/doracled/cmd/upgrade_oracle.go

```diff
@@ -0,0 +1,116 @@
+package cmd
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	oracletypes "github.com/medibloc/panacea-core/v2/x/oracle/types"
+	"github.com/medibloc/panacea-doracle/client/flags"
+	log "github.com/sirupsen/logrus"
+	"github.com/tendermint/tendermint/libs/os"
+	"io"
+
+	"github.com/edgelesssys/ego/enclave"
+	"github.com/medibloc/panacea-doracle/panacea"
+	"github.com/spf13/cobra"
+)
+
+func upgradeOracleCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "upgrade-oracle",
+		Short: "Upgrade an oracle",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			conf, err := loadConfigFromHome(cmd)
+			if err != nil {
+				return err
+			}
+			ctx := context.Background()
+
+			// if node key exists, return error.
+			nodePrivKeyPath := conf.AbsNodePrivKeyPath()
+			if os.FileExists(nodePrivKeyPath) {
```

Comment:
In general, there is an existing node key for oracle registration (or for previous version).
That is, most user would encounter this error. Is it ok? Or overwrite the existing node key when upgrading?
One more option is to use separate dir for node key depending on uniqueID as you suggested before.

Reply:
At upgrade time, the HOME path will probably be set differently.
Because we can't run multi doracles in the same path.
So, this part uses the existing `ReigsterOracle` function as it is.